### PR TITLE
[doc] Update earlgrey documentation for ast clock/reset mapping

### LINKED ISSUE #6798: [doc] Missing documentation on the relationship between AST clocks/resets and chip clocks/resets
For example, to which clock the AST entropy clock input is connected? Is AST ADC reset input connected to vcaon_pok or to vcmain_pok?

### PR DESCRIPTION
**Pull request by tjaychen**
- fixes #6798

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

File: hw/top_earlgrey/doc/design/index.md [hunk 1]
```diff
@@ -21,18 +21,59 @@ This section also contains a brief overview of some of the features of the final
...
+- vcaon_pok: The always on domain of the system is ready.
+- vcmain_pok: The main operating domain of the system is ready.
+
+When one of these power-okay signals drop, the corresponding domain in `top_earlgrey` is reset.
+Please refer to [reset manager]({{< relref "hw/ip/rstmgr/doc/" >}})) for more details.
 Resets throughout the design are asynchronous active low as per the Comportability specification.
-Other resets may be generated internally via the alert responder, watchdog timer, etc., and other resets for subunits may be disseminated.
-These will be detailed in this section over time.
+
+Once reset, the reset vector begins in ROM, whose job is to validate code in the emulated e-flash before jumping to it.
```
Comment (zi-v) [inline hw/top_earlgrey/doc/design/index.md:43]: nit: emulated e-flash -> embedded flash

**Review by zi-v — COMMENTED** [review]
LGTM

File: hw/top_earlgrey/doc/design/index.md (hunk 1 above)
Comment (tjaychen) [reply hw/top_earlgrey/doc/design/index.md:43]: ah yes good catch.

Comment (tjaychen) [reply hw/top_earlgrey/doc/design/index.md]: updated

**Review by msfschaffner — APPROVED** [review]

**Pull request merged**